src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -289,6 +289,60 @@ internal ValueTask UnsubscribeAsync(int sid)
         return default;
     }
 
+    private static NatsOpts ReadUserInfoFromConnectionString(NatsOpts opts)
+    {
+        var first = true;
+
+        var natsUris = opts.GetSeedUris();
```

Comment:
GetSeedUris could return a random one from the list. we should check the first one provided. edit: maybe add an optional parameter to always get the original order